api/envoy/config/filter/http/router/v2/router.proto

```diff
@@ -26,4 +26,15 @@ message Router {
   // an upstream request. Presuming retries are configured, multiple upstream
   // requests may be made for each downstream (inbound) request.
   repeated envoy.config.filter.accesslog.v2.AccessLog upstream_log = 3;
+
+  // If set, the specified header will be added to upstream requests with a timestamp
+  // (in milliseconds) when Envoy starts decoding downstream's request headers.
+  //
+  // Example:
+  //
+  // .. code-block:: yaml
+  //
+  //   start_timestamp_header: x-request-start
+  //
+  string start_timestamp_header = 4;
```

Comment:
Can we reuse the existing user defined headers mechanism for this? E.g. you could configure `x-request-start` with value `%START_TIMESTAMP%`?

Reply:
Oh yes, +1. That seems better.

Reply:
@htuch thanks for the hint! Yes, I think that's the right direction (I should went to that direction in the first place, since apparently `RequestInfo` has that useful `.startTime()` 😔). 

I added the corresponding changes using that approach. Thanks!

Reply:
That's cleaner, thanks for suggesting!